registry/dogu_etcd.go

```diff
@@ -0,0 +1,241 @@
+package registry
+
+import (
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/pkg/errors"
+)
+
+// combinedEtcdDoguRegistry was added to provide backward compatibility in the dogu registry.
+// All writing actions use v1 and v2 registry
+// All reading actions use v2 registry, v1 registry if key not found in v2.
+type combinedEtcdDoguRegistry struct {
+	v1DoguRegistry DoguRegistry
+	v2DoguRegistry DoguRegistry
+}
+
+func newCombinedEtcdDoguRegistry(client *resilentEtcdClient, pathV1 string, pathV2 string) *combinedEtcdDoguRegistry {
+	formatProviderV1 := &core.DoguJsonV1FormatProvider{}
+	formatProviderV2 := &core.DoguJsonV2FormatProvider{}
+
+	return &combinedEtcdDoguRegistry{
+		v1DoguRegistry: &etcdDoguRegistry{
+			pathV1,
+			client,
+			formatProviderV1,
+		},
+		v2DoguRegistry: &etcdDoguRegistry{
+			pathV2,
+			client,
+			formatProviderV2,
+		},
+	}
+}
+
+// Enable enables a specific dogu in the backend. The method will create the
+// current key to the dogu path
+// Enables the dogu in v1 as well as in v2 registry.
+func (reg *combinedEtcdDoguRegistry) Enable(dogu *core.Dogu) error {
+	err := reg.v1DoguRegistry.Enable(dogu)
+	if err != nil {
+		return errors.Wrap(err, "could not write to v1 registry")
+	}
+
+	err = reg.v2DoguRegistry.Enable(dogu)
+	if err != nil {
+		return errors.Wrap(err, "could not write to v2 registry")
+	}
+
+	return nil
+}
+
+// Register registers the dogu at the registry backend
+// Registeres the dogu in v1 as well as in v2 registry.
```

Comment:
Typo while we are at it:
```suggestion
// Registers the dogu in v1 as well as in v2 registry.
```

Reply:
Done